libsysbox/syscont/spec.go

```diff
@@ -569,6 +569,21 @@ func cfgSysboxMounts(spec *specs.Spec) {
 		return m1.Destination == m2.Destination
 	})
 
+	// If the container's rootfs is read-only, then sysbox mounts of /sys and
+	// below should also be read-only.
+	if spec.Root.Readonly {
+		tmpMounts := []specs.Mount{}
+		rwOpt := []string{"rw"}
+		for _, m := range sysboxMounts {
+			if strings.HasPrefix(m.Destination, "/sys") {
```

Comment:
If we want to be fully consistent with this, then we should also mount `/proc/swaps` and `/proc/uptime` in RO mode, even though they are read-only resources ...

```
│ ├─/proc/swaps                                                                                      sysboxfs[/proc/swaps]  fuse     rw,nosuid,nodev,relatime,user_id=0,group_id=0,default_permissions,allow_other
│ ├─/proc/sys                                                                                        sysboxfs[/proc/sys]    fuse     rw,nosuid,nodev,relatime,user_id=0,group_id=0,default_permissions,allow_other
│ └─/proc/uptime                                                                                     sysboxfs[/proc/uptime] fuse     rw,nosuid,nodev,relatime,user_id=0,group_id=0,default_permissions,allow_other
```

Reply:
Yes the code does this already,  but does it in line 618.